airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml

```diff
@@ -427,33 +455,79 @@ definitions:
         type: string
         enum: [OAuthAuthenticator]
       client_id:
+        title: Client ID
+        description: The OAuth client ID
         type: string
+        examples:
+          - "{{ config['client_id }}"
+          - "{{ config['credentials']['client_id }}"
       client_secret:
+        title: Client Secret
+        description: The OAuth client secret
         type: string
+        examples:
+          - "{{ config['client_secret }}"
+          - "{{ config['credentials']['client_secret }}"
       refresh_token:
+        title: Refresh Token
+        description: Credential artifact used to get a new access token
         type: string
+        examples:
+          - "{{ config['refresh_token'] }}"
+          - "{{ config['credentials]['refresh_token'] }}"
       token_refresh_endpoint:
+        title: Token Refresh Endpoint
+        description: Endpoint to send a POST request in order to get a new access token
         type: string
+        examples:
+          - https://connect.squareup.com/oauth2/token
       access_token_name:
+        title: Access Token Field Name
+        description: The name of the field to extract the access token from in the token refresh response
         type: string
         default: "access_token"
+        examples:
+          - access_token
       expires_in_name:
+        title: Expires_in Field Name
+        description: The name of the field the extract the number of seconds the access token will be valid from the token refresh response
         type: string
         default: "expires_in"
+        examples:
+          - expires_in
       grant_type:
+        title: Grant Type
+        description: How the access token is granted
         type: string
         default: "refresh_token"
+        examples:
+          - refresh_token
```

Comment:
There are [other types](https://aaronparecki.com/oauth-2-simplified/#authorization), but they're not easily supported.

For example, it should be possible to use `grant_type=authorization_code`, use the `refresh_token` field as the authorization code, and add the required fields to the `refresh_request_body`:
example:
```
client_id: "{{ config['client_id'] }}" 
client_secret: "{{ config['client_secret'] }}"
refresh_token: "{{ config['authorization_code'] }}"
grant_type: "authorization_code"
refresh_request_body:
  code: "{{ config['authorization_code'] }}"
```

This however feels pretty hacky and isn't something I tested. I would prefer we simplify the interface by removing the fields and explicitly only allowing refresh_token grant type/